src/commands/app/test.js

```diff
@@ -10,40 +10,229 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
+const { runScript } = require('../../lib/app-helper')
 const { flags } = require('@oclif/command')
-const appHelper = require('../../lib/app-helper')
 const BaseCommand = require('../../BaseCommand')
+const chalk = require('chalk')
+const aioLogger = require('@adobe/aio-lib-core-logging')('@adobe/aio-cli-plugin-app:test', { provider: 'debug' })
+const path = require('path')
 
 class Test extends BaseCommand {
   async run () {
     const { flags } = this.parse(Test)
-    // some things we could do here:
-    // test configurations, ie remote-actions deployed and called from local
-    // this just runs package.json scripts.test, we could also check that this is in fact an aio app project
-    const command = flags.e2e ? 'e2e' : 'test'
-    try {
-      await appHelper.runPackageScript(command, process.cwd())
-    } catch (e) {
-      return this.error(e.message, { exit: e.exitCode })
+    let { all, unit, e2e, action, extension } = flags
+
+    // 'all' overrides the setting of either the unit or e2e flag
+    if (all) {
+      unit = true
+      e2e = true
+    } else if (!unit && !e2e) {
+      // 'all' not set; we check if neither is set, and default to 'unit'
+      unit = true
     }
+
+    const buildConfigs = this.getAppExtConfigs({ extension })
+    aioLogger.debug(`run buildConfigs:${JSON.stringify(buildConfigs, null, 2)}`)
+
+    const totalResults = []
+    let exitCode = 0
+    for (const [extensionName, extensionConfig] of Object.entries(buildConfigs)) {
+      const results = await this.runExtensionTest(extensionName, extensionConfig, { unit, e2e, action })
+      totalResults.push(...results)
+
+      const failures = results.filter(result => !result.passed)
+      if (failures.length > 0) {
+        exitCode += failures.length
```

Comment:
is this a standard ? I thought exit codes have special meaning otherwise let's just return 1 ?

Reply:
not a standard, yeah I will change it